hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java

```diff
@@ -650,6 +650,10 @@ public void terminate(String reason, Exception cause, boolean clearMetrics, bool
         if (worker.entryReader.isAlive()) {
           worker.entryReader.interrupt();
         }
+      } else {
+        //If worker is already stopped but there was still entries batched,
+        //we need to clear buffer used for non processed entries
+        worker.clearWALEntryBatch();
```

Comment:
We will not do this for the above condition branch?

Reply:
We don't want to clear entry batch if shipper thread is still running, because it may race for the bufferUsage update. However, I realised we need to go through the branch first to interrupt it, got confused by an additional loop calling `stopWorker` prior to this one. I'm fixing this on a following commit.